### PR TITLE
Make NuGet.Credentials XPLAT, cleanup the dependency graph

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug
Fixes: 
Regression: No  
If Regression then when did it last work:   
If Regression then how are we preventing it in future:   

## Fix
Details: 
As part of the cross-plat auth plugin work, we'd need NuGet.Credentials to be XPLAT. 
I've changed around a couple of dependencies.  NuGet.Credentials in reality needs only NuGet.Configuration, but I've made it depend on Nuget.Protocol, as that's where the Plugin code is. 

I've also cleaned up a bunch of the redundant dependencies.
Extra dependencies to NuGet.Frameworks and newtonsoft.json.

I will comment on each one, why each change is made here in the PR.

I want to do this change early, because it's isolated. It'll also make it easier to review other PRs with the needless fluff from 10 extra files. 

## Testing/Validation
Tests Added: No  
Reason for not adding tests:  
Validation done:  
Built. Existing tests are the only thing that need to work.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Commands/NuGet.Commands.csproj [hunk 1]
```diff
@@ -25,7 +25,6 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\NuGet.Frameworks\NuGet.Frameworks.csproj" />
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/NuGet.Commands.csproj:28]: NuGet.Configuration => NuGet.Common => NuGet.Frameworks

File: src/NuGet.Core/NuGet.LibraryModel/NuGet.LibraryModel.csproj [hunk 2]
```diff
@@ -21,7 +21,6 @@
   <ItemGroup>
     <ProjectReference Include="..\NuGet.Common\NuGet.Common.csproj" />
     <ProjectReference Include="..\NuGet.Versioning\NuGet.Versioning.csproj" />
-    <ProjectReference Include="..\NuGet.Frameworks\NuGet.Frameworks.csproj" />
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.LibraryModel/NuGet.LibraryModel.csproj:24]: NuGet.Common => NuGet.Frameworks

File: src/NuGet.Core/NuGet.Packaging.Core/NuGet.Packaging.Core.csproj [hunk 3]
```diff
@@ -20,7 +20,6 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <ProjectReference Include="$(NuGetCoreSrcDirectory)NuGet.Frameworks\NuGet.Frameworks.csproj" />
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging.Core/NuGet.Packaging.Core.csproj:23]: NuGet.Common => NuGet.Frameworks

File: src/NuGet.Core/NuGet.ProjectModel/NuGet.ProjectModel.csproj [hunk 4]
```diff
@@ -22,12 +22,7 @@
     <ProjectReference Include="..\NuGet.DependencyResolver.Core\NuGet.DependencyResolver.Core.csproj" />
   </ItemGroup>
 
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net46' ">
-    <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersionCore)" />
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.ProjectModel/NuGet.ProjectModel.csproj:26]: Dependency.Resolver.Core => NuGet.Protocol =>NuGet.Packaging => Newtonsoft.Json

File: src/NuGet.Core/NuGet.Protocol/NuGet.Protocol.csproj [hunk 5]
```diff
@@ -44,7 +44,6 @@
   </ItemGroup>
 
   <ItemGroup Condition=" '$(TargetFramework)' == 'net46' ">
-    <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersionCore)" />
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Protocol/NuGet.Protocol.csproj:47]: NuGet.Packaging => Newtonsoft.Json

File: src/NuGet.Clients/NuGet.Credentials/PluginException.cs [hunk 6]
```diff
@@ -17,12 +17,12 @@ public class PluginException : Exception
         public PluginException(string message) : base(message) { }
 
         public PluginException(string message, Exception inner) : base(message, inner) { }
-
+#if NET46
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.Credentials/PluginException.cs]: nit: change it to IS_DESKTOP

**Review by jainaashish — APPROVED** [review]

**Review by rohit21agrawal — APPROVED** [review]

**Pull request merged**